chemprop/models/model.py

```diff
@@ -33,7 +34,16 @@ def __init__(self, args: TrainArgs):
         if self.classification or self.multiclass:
             self.no_training_normalization = args.loss_function in ['cross_entropy', 'binary_cross_entropy']
 
-        self.output_size = args.num_tasks
+        self.is_atom_bond_targets = args.is_atom_bond_targets
+
+        if self.is_atom_bond_targets:
+            self.atom_targets, self.bond_targets = args.atom_targets, args.bond_targets
+            self.atom_constraints, self.bond_constraints = args.atom_constraints, args.bond_constraints
+            self.device = args.device
```

Comment:
Why is this separate device setting necessary?

Reply:
It would be used in L70 and L79 in train/predict.py.

Reply:
Why doesn’t the model.to(device) in run_training or the device argument in load_checkpoint cover it like it does for molecule targets?

Reply:
`device` is not an attribute of nn.Module. A similar problem has been raised [here](https://github.com/pytorch/pytorch/issues/7460). As mentioned by Kani (in the comments), if all the parameters are on a single device, one could use `next(model.parameters()).device` to get the device of the model. I would try this way to remove the separate device setting.